### PR TITLE
updated calendar to show next week on Sundays

### PR DESCRIPTION
**Pull request by Anasailon**
Added a check for sundays to display the following week on sundays. For this the date objct that was passed to the update calendar function had to be changed from start of current Type to start of day

### CODE REVIEW

File: cmd/ical-relay/templates/calendar.html [hunk 1]
```diff
@@ -111,8 +111,12 @@
                 calendarEnd = date.add(4, "week").day(6);
             }else{
                 document.getElementById("current-date").innerHTML = "KW"+currentDate.format(" w MMMM");
-
+                console.log(date);
+                console.log(date.day());
```
Comment (nilsfast) [inline cmd/ical-relay/templates/calendar.html:115]: please remove useless console logs

**Review by nilsfast — CHANGES_REQUESTED** [review]

**Comment by robske110** [comment]
This displays the wrong calendar Week in the title and (likely?) breaks the month view.

Also, I don't know if introducing this statically (without configuration) makes sense, as this is intended to be a general-purpose calendar viewer. Some calendars do have events on sunday.
Potentially a better solution would be to skip to the next week if no events in the current week remain.

File: cmd/ical-relay/templates/calendar.html [hunk 2]
```diff
@@ -111,12 +111,12 @@
...
             }else{
                 document.getElementById("current-date").innerHTML = "KW"+currentDate.format(" w MMMM");
-
                 calendarStart = date.startOf("week").subtract(1, "day");
+                if(date.day() === 0){
+                    calendarStart = calendarStart.add(1,"week")
+                }
                 calendarEnd = calendarStart.add(1, "week");
             }
-            console.log(calendarStart);
```
Comment (robske110) [inline cmd/ical-relay/templates/calendar.html:118]: These logs are somewhat useful for debugging and checking what was interpreted from the current date input (location hash). I would keep them.
They could be made more useful by converting them to output something like "Showing events from cStart to cEnd", but that's not necessary.

**Review by robske110 — CHANGES_REQUESTED** [review]

File: cmd/ical-relay/templates/calendar.html [hunk 3]
```diff
@@ -111,12 +111,12 @@
                 calendarEnd = date.add(4, "week").day(6);
             }else{
                 document.getElementById("current-date").innerHTML = "KW"+currentDate.format(" w MMMM");
-
                 calendarStart = date.startOf("week").subtract(1, "day");
+                if(date.day() === 0){
+                    calendarStart = calendarStart.add(1,"week")
+                }
```
Comment (robske110) [inline cmd/ical-relay/templates/calendar.html:117]: Because this modifies the timeframe outlined by calendarStart and calendarEnd to not include currentDate (selectedDate), this causes the wrong calendarWeek to be displayed in the title and also the location hash to be wrong (although it is debatable if this is bad).
I think the best solution is to just move this check into the beginning of the function setSelectedDate, which also eliminates the need for the change on line 144.

File: cmd/ical-relay/templates/calendar.html [hunk 4]
```diff
@@ -141,7 +141,7 @@
                     currentType = "month";
                 }
             }
-            currentDate = location.hash ? dayjs(location.hash.substring(1)) : dayjs().startOf(currentType);
+            currentDate = location.hash ? dayjs(location.hash.substring(1)) : dayjs().startOf('day');
```
Comment (robske110) [inline cmd/ical-relay/templates/calendar.html:144]: This change should be removed when the check in updateCalendar is moved to setSelectedDate.

**Review by robske110 — CHANGES_REQUESTED** [review]

File: cmd/ical-relay/templates/calendar.html (hunk 2 above)
Comment (Anasailon) [reply cmd/ical-relay/templates/calendar.html:118]: chef hat gesagt soll weg

Comment (robske110) [reply cmd/ical-relay/templates/calendar.html:118]: I think @nilsfast was talking about the 
```
console.log(date);
console.log(date.day());
```
you added, not these two ones which were already there. 
@nilsfast please clarify

Comment (JM-Lemmi) [reply cmd/ical-relay/templates/calendar.html:118]: chef = @nilsfast

Comment (nilsfast) [reply cmd/ical-relay/templates/calendar.html:118]: undeclared use of `chef` use `:=` to declare

Comment (nilsfast) [reply cmd/ical-relay/templates/calendar.html:118]: @robske110 correct. Keep the ln. 118 @Anasailon, I think logging the current date in prod is a bit useless

**Review by robske110 — CHANGES_REQUESTED** [review]
Now I think we've overcomplicated this a bit. Sorry if I was confusing in my last review.
To be clear, I would propose to just reduce the changes to this:
image
That should work even with proper updating of the location in the url.

File: cmd/ical-relay/templates/calendar.html (hunk 2 above)
Comment (robske110) [reply cmd/ical-relay/templates/calendar.html:118]: and also ln. 119 :)

File: cmd/ical-relay/templates/calendar.html [hunk 5]
```diff
@@ -107,16 +107,18 @@
             if(currentType === "month"){
                 document.getElementById("current-date").innerHTML = currentDate.format("MMMM YYYY");
 
-                calendarStart = date.day(0);
+                calendarStart = date.startOf('month').day(0);
```
Comment (robske110) [inline cmd/ical-relay/templates/calendar.html:110]: This change can go now. (Not needed anymore)

File: cmd/ical-relay/templates/calendar.html [hunk 6]
```diff
@@ -107,16 +107,18 @@
...
                 calendarEnd = date.add(4, "week").day(6);
             }else{
-                document.getElementById("current-date").innerHTML = "KW"+currentDate.format(" w MMMM");
-
                 calendarStart = date.startOf("week").subtract(1, "day");
                 calendarEnd = calendarStart.add(1, "week");
+
+                document.getElementById("current-date").innerHTML = "KW"+calendarStart.add(1,"day").format(" w MMMM");
```
Comment (robske110) [inline cmd/ical-relay/templates/calendar.html:116]: This change can also be removed.

File: cmd/ical-relay/templates/calendar.html [hunk 7]
```diff
@@ -107,16 +107,18 @@
...
             }
+
```
Comment (robske110) [inline cmd/ical-relay/templates/calendar.html:118]: unrelated whitespace change

File: cmd/ical-relay/templates/calendar.html [hunk 8]
```diff
@@ -107,16 +107,18 @@
...
             console.log(calendarStart);
             console.log(calendarEnd);
+
```
Comment (robske110) [inline cmd/ical-relay/templates/calendar.html:121]: unrelated whitespace change

File: cmd/ical-relay/templates/calendar.html [hunk 9]
```diff
@@ -141,13 +143,19 @@
...
+            }
+            updateCalendar(currentDate.startOf(currentType));
         }
         addEventListener("hashchange", onHashChange)
         onHashChange();
 
         function setSelectedDate(date) {
+            if(date.day() === 0){
+                    date = date.add(1,"day")
+                }
```
Comment (robske110) [inline cmd/ical-relay/templates/calendar.html:158]: nitpick: whitespace between 1, "day" and indentation of }

**Review by robske110 — CHANGES_REQUESTED** [review]
Almost there now :)

**Comment by Anasailon** [comment]
> Now I think we've overcomplicated this a bit. Sorry if I was confusing in my last review. To be clear, I would propose to just reduce the changes to this: image That should work even with proper updating of the location in the url.

was resolved by moving to check in the setcurrentdate function and also before currentdate is passed to updatecalendar

**Review by frereit — APPROVED** [review]
nice work 😄

**Review by robske110 — APPROVED** [review]
Looks good to me now 👍 
Was a bit more intricate than initially thought.
Would suggest to squash-merge this.

**Review by nilsfast — APPROVED** [review]
LGTM
